src/lib/update-alternatives/UI/main_dialog.rb

```diff
@@ -74,6 +85,18 @@ def map_alternatives_items
       end
     end
 
+    def apply_filters
+      if @one_choice_filter
+        @alternatives_list.each_with_index.select { |v, _i| v.choices.length > 1 }
+      else
+        @alternatives_list.each_with_index
+      end
+    end
+
+    def filters
+      CheckBox(Id(:one_choice_filter), Opt(:notify), "Filter alternatives with one choice", true)
```

Comment:
here is problem, that you have at two places tight default values for `one_choice_filter` so if you change it in initialize when you e.g. decide you want different default, then bug immediately occur as it is still checked. Better to use here already value, so it is explicitly coupled `CheckBox(Id(:one_choice_filter), Opt(:notify), "Filter alternatives with one choice", @one_choice_filter)`